### PR TITLE
B-17235 Okta API Call to Edit

### PR DESCRIPTION
**Pull request by danieljordan-caci**
## Summary

When a user goes to make changes to their Okta profile, they input updated values into the form.
Upon clicking "Save", this will call the Okta Users API and update their profile.
MilMove will then reflect changed values in state as well as calling a GET request to the Okta API on page refresh to ensure updated values are shown and not values in Access Token (which can be dated).

## Verification Steps for Reviewers

Upon logging into Customer MilMove, please ensure your Okta profile values are up to date and accurate.
Saving new values in the form should send you back to the profile page, display a success banner, and you should see the updated values in the form viewer.

### Setup to Run the Code

- switch to branch
- `make client_run`
- `make server_run`
- Once built and compiled, log into MilMove using your Okta credentials as a customer (please note that if you are a part of the office OR admin groups, you will not be able to access the application and will need to create a new user or remove yourself from those groups in Okta).
- Navigate to the profile section
- Update your Okta values
- Ensure changes are accurate

### Frontend

- Updated EditOktaInfo component to reflect Okta values stored in state
- Updated Redux store with Okta actions that are fired when `SET_OKTA_USER` action is called
- Reflected changes in swagger UI and .yaml files

### Backend

- Added GET and POST requests to `okta_profile.go` handler

### CODE REVIEW

**Comment by deandreJones** [comment]
image
is it expected that the first time they login that the okta profile section is empty?

**Comment by danieljordan-caci** [comment]
> image is it expected that the first time they login that the okta profile section is empty?

Did you log in via Okta or through the "Local Sign-in" option? When you log in with the Local Sign-in option, it does not load Okta so this would be what would be shown when you do that.

**Comment by deandreJones** [comment]
just tried again, and i am getting same results, still getting not provided when logging in through okta

**Comment by danieljordan-caci** [comment]
> just tried again, and i am getting same results, still getting not provided when logging in through okta

For some reason I had the wrong return values in my post request in swagger. I have it working on my end now, but let me know if it passes for you.

**Comment by cameroncaci** [comment]
the okta-api-key secret has been added to all environments correct? Dev, stg, prd, exp, demo, loadtest?

File: pkg/cli/auth.go [hunk 1]
```diff
@@ -24,6 +24,18 @@ const (
...
+	//RA Summary: gosec - G101 - Password Management: Hardcoded Password
+	//RA: This line was flagged because of use of the word "secret"
+	//RA: This line is used to identify the name of the flag. OktaApiKeyFlag is the Okta API Key Flag.
+	//RA: This value of this variable does not store an application secret.
+	//RA Developer Status:
+	//RA Validator Status:
+	//RA Validator:
+	//RA Modified Severity:
+	// #nosec G101
+	OktaAPIKeyFlag string = "okta-api-key"
```
Comment (cameroncaci) [inline pkg/cli/auth.go]: Is this nosec comment necessary?

File: pkg/handlers/authentication/okta/provider.go [hunk 2]
```diff
@@ -116,7 +123,7 @@ func (op *Provider) AuthorizationURL(r *http.Request) (*Data, error) {
 	params := authURL.Query()
 	// Add the nonce and scope to the URL when getting ready to redirect to the login URL
 	params.Add("nonce", state)
-	params.Set("scope", "openid profile email")
+	params.Set("scope", "openid profile email okta.myAccount.profile.read")
```
Comment (cameroncaci) [inline pkg/handlers/authentication/okta/provider.go]: Question out of ignorance here - I know we were initially lookign throgh scopes for OIDC usage. Is this scope still necessary if using API key?

File: pkg/handlers/authentication/okta_auth_code_flow.go [hunk 3]
```diff
@@ -82,6 +82,7 @@ func exchangeCode(code string, r *http.Request, appCtx appcontext.AppContext, pr
 	q.Add("grant_type", "authorization_code")
 	q.Set("code", code)
 	q.Add("redirect_uri", provider.GetCallbackURL())
+	q.Add("scope", "openid email profile")
```
Comment (cameroncaci) [inline pkg/handlers/authentication/okta_auth_code_flow.go:85]: If that new scope is necessary, would it be needed here too?

**Review by cameroncaci — CHANGES_REQUESTED** [review]
Quick tidbit

File: pkg/cli/auth.go (hunk 1 above)
Comment (deandreJones) [inline pkg/cli/auth.go:37]: does this OktaAPIKeyFlag also need to be added to CircleCi config?

Comment (danieljordan-caci) [reply pkg/cli/auth.go]: I could not pass the tests if I didn't have that there, so I'm assuming yes?

File: pkg/handlers/authentication/okta/provider.go (hunk 2 above)
Comment (danieljordan-caci) [reply pkg/handlers/authentication/okta/provider.go]: It is not, I will correct this. Nice catch!

File: pkg/cli/auth.go (hunk 1 above)
Comment (danieljordan-caci) [reply pkg/cli/auth.go:37]: If it is needed for CircleCI to pass tests, then probably so. I'm not sure how to do that though.

File: pkg/handlers/authentication/okta/provider.go (hunk 2 above)
Comment (danieljordan-caci) [reply pkg/handlers/authentication/okta/provider.go]: Corrected and pushed

**Comment by danieljordan-caci** [comment]
> the okta-api-key secret has been added to all environments correct? Dev, stg, prd, exp, demo, loadtest?

Yes it has! I've added it to all environments in chamber.

**Review by deandreJones — APPROVED** [review]

**Review by j-Rosenthal — APPROVED** [review]
LGTM :-)

**Review by cameroncaci — APPROVED** [review]

**Pull request merged**